src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -239,6 +239,54 @@ export class ExposureNotificationService {
     await this.recordKeySubmission();
   }
 
+  calculateNeedsSubmission(exposureStatus: ExposureStatus, today: Date): boolean {
+    if (exposureStatus.type !== ExposureStatusType.Diagnosed) return false;
+
+    const cycleEndsAt = new Date(exposureStatus.cycleEndsAt);
+    // We're done submitting keys
+    // This has to be based on UTC timezone https://github.com/cds-snc/covid-shield-mobile/issues/676
+    if (daysBetweenUTC(today, cycleEndsAt) <= 0) return false;
+
+    const submissionLastCompletedAt = exposureStatus.submissionLastCompletedAt;
+    if (!submissionLastCompletedAt) {
+      return true;
+    }
+
+    const lastSubmittedDay = new Date(submissionLastCompletedAt);
+
+    // This has to be based on UTC timezone https://github.com/cds-snc/covid-shield-mobile/issues/676
+    return daysBetweenUTC(lastSubmittedDay, today) > 0;
```

Comment:
Given we're looking to return a Boolean here - would be good to double check what happens when bad data is given to daysBetweenUTC to see what it returns.

Reply:
> Given we're looking to return a Boolean here - would be good to double check what happens when bad data is given to daysBetweenUTC to see what it returns.

I've added in some checks to ensure that the date values being used are actually dates.